docs/testplan/SmartSwitch-PMON-testplan.md

```diff
@@ -0,0 +1,939 @@
+| 1.14 | Check the cold startup of the DPUs and SWITCH       |  To verify the "chassisd", "chassis instance o database" are created and the admin state of the DPUs as defined in the config_db.json are applied properly |
+| 1.15 |  Check the startup config CLI       |  To verify the "config chassis modules startup DPUx" CLI does startup the DPU|
+| 1.16 | Check the shutdown config CLI        |   To verify the "config chassis modules shutdown DPUx" CLI does shutdown the DPU|
+| 1.17 | Check the reboot config CLI       |  To verify the "config chassis modules reboot SWITCH " CLI does reboot the SWITCH|
+
+## Test Cases
+
+
+### 1.1 Check SmartSwitch specific ChassisClass APIs
+
+#### Steps
+ * Execute the following CLIs on SmartSwitch
+ * get_dpu_id(self, name):
+    * Provide DPU0-DPU7
+    * This API should return 1-8 for DPU0-DPU7
+ * is_smartswitch(self):
+    * This API should return True
+ * get_module_dpu_data_port(self, index):
+    * For index: 1 will return the dup0 port association which is "Ethernet224: Ethernet0" where the string left of ":" (Ethernet224) is the NPU port and the string right of ":" (Ethernet0) is the DPU port.
+ 
+
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch:
+    get_dpu_id(self, DPU3)
+    Output: 4
+
+    is_smartswitch(self):
+    Output: True
+
+    get_module_dpu_data_port(self, DPU0):
+    Output: "Ethernet224: Ethernet0"
+
+```
+#### Pass/Fail Criteria
+ *  The test result is a pass if the return value matches the expected value as shown in the "steps" and "Sample Output".
+
+
+### 1.2 Check modified ChassisClass APIs for SmartSwitch
+
+#### Steps
+ * is_modular_chassis(self):
+    * Should return False
+ * get_num_modules(self):
+    * Should return number of DPUs + 1 switch
+ * get_module(self, index):
+    * Make sure for each index this API returns an object and has some content and not None
+ * get_all_modules(self):
+    * This should return a list of items
+ * get_module_index(self, module_name):
+    * Given the module name say “DPU0” should return the index of it “1”
+ 
+
+#### Verify in
+ * Switch
+   
+#### Sample Output
+```
+On Switch:
+    is_modular_chassis(self):
+    Output: False
+
+    get_num_modules(self):
+    Output: number of DPUs + 1
+
+    get_module(self, DPU0):
+    Output: DPU0 object
+
+    get_all_modules(self):
+    Output: list of objects (one per DPU + 1 switch object)
+
+    get_module_index(self, DPU0):
+    Output: could be any value from 0 to modules count -1
+```
+#### Pass/Fail Criteria
+ * The test result is a pass if the return value matches the expected value as shown in the "steps" and "Sample Output"
+
+### 1.3 Check DpuModule APIs for SmartSwitch
+
+#### Steps
+ * get_dpu_id(self):
+    * Should return ID of the DpuModule Ex: 1 on DPU0
```

Comment:
Now when the agreed to not implement switch module the ID of the DPU0 should be 0

Reply:
This is a huge test effort .  Let us stay as such.